src/react/next-architecture/domain/business/accounts.test.tsx

```diff
@@ -0,0 +1,294 @@
+    const metricsAdapter = new MockedMetricsAdapter();
+    metricsAdapter.listAccountsLatestUsedCapacity = jest
+      .fn()
+      .mockImplementation(async () => {
+        return Promise.reject('error');
+      });
+    //E
+    const { result } = renderHook(
+      () => useListAccounts({ accountsAdapter, metricsAdapter }),
+      { wrapper },
+    );
+    //V
+    expect(result.current.accounts).toStrictEqual({
+      status: 'success',
+      value: [
+        {
+          ...ACCOUNT,
+          usedCapacity: {
+            status: 'error',
+            title: 'An error occurred when fetching metrics',
+            reason: 'error',
+          },
+        },
+        {
+          ...NEWLY_CREATED_ACCOUNT,
+          usedCapacity: {
+            status: 'error',
+            title: 'An error occurred when fetching metrics',
+            reason: 'error',
+          },
+        },
+      ],
+    });
+  });
+});
+
+describe.skip('useAccountLatestUsedCapacity', () => {
+  it('should return the metrics for an account', () => {
+    //S
+    const metricsAdapter = new MockedMetricsAdapter();
+    //E
+    const { result } = renderHook(
+      () =>
+        useAccountLatestUsedCapacity({
+          metricsAdapter,
+          accountCanonicalId: ACCOUNT_CANONICAL_ID,
+        }),
+      {
+        wrapper,
+      },
+    );
+    //V
+    expect(result.current.usedCapacity).toStrictEqual({
+      status: 'success',
+      value: ACCOUNT_METRICS,
+    });
+  });
+  it('should return error in cause of fetching metrics failed', () => {
+    //S
+    const metricsAdapter = new MockedMetricsAdapter();
+    metricsAdapter.listAccountsLatestUsedCapacity = jest
+      .fn()
+      .mockImplementation(async () => Promise.reject('error'));
+    //E
+    const { result } = renderHook(
+      () =>
+        useAccountLatestUsedCapacity({
+          metricsAdapter,
+          accountCanonicalId: ACCOUNT_CANONICAL_ID,
+        }),
+      {
+        wrapper,
+      },
+    );
+    //V
+    expect(result.current.usedCapacity).toStrictEqual({
+      status: 'error',
+      title: 'An error occurred when fetching the metrics',
+      reason: 'error',
+    });
+  });
+  it('should not be executed if the account metrics has been retrieved', async () => {
+    //S
+    const metricsAdapter = new MockedMetricsAdapter();
+    const accountsAdapter = new MockedAccountsAdapter();
+    //E
+    const { result: resultAccounts, waitFor } = renderHook(
+      () =>
+        useListAccounts({
+          metricsAdapter,
+          accountsAdapter,
+        }),
+      {
+        wrapper,
+      },
+    );
+    await waitFor(() => resultAccounts.current.accounts.status === 'success');
+    const { result } = renderHook(
+      () =>
+        useAccountLatestUsedCapacity({
+          metricsAdapter,
+          accountCanonicalId: ACCOUNT_CANONICAL_ID,
+        }),
+      {
+        wrapper,
+      },
+    );
+    //V
+    expect(result.current.usedCapacity.status).toBe('idle');
```

Comment:
```suggestion
    expect(result.current.usedCapacity.status).toBe('success');
```

Reply:
Yes. I'll also check that the status won't go through `loading`.